File: kern/vfdd.py
"""Kernel-side vFDD block API (host model for S7a)."""

from __future__ import annotations

from plover_vm.memory.vfdd import VirtualFdd


class VfddDriver:
    def __init__(self, dev: VirtualFdd) -> None:
        self.dev = dev

    def read_sector(self, n: int) -> bytes:
        return self.dev.read_sector(n)

    def write_sector(self, n: int, data: bytes) -> None:
        self.dev.write_sector(n, data)

File: crates/plover_os/src/vfdd.rs
use plover_copro::vfdd::{VirtualFdd, VfddError};

pub struct VfddDriver {
    dev: VirtualFdd,
}

impl VfddDriver {
    pub fn new(dev: VirtualFdd) -> Self {
        Self { dev }
    }

    pub fn read_sector(&self, n: usize) -> Result<Vec<u8>, VfddError> {
        self.dev.read_sector(n)
    }

    pub fn write_sector(&self, n: usize, data: &[u8]) -> Result<(), VfddError> {
        self.dev.write_sector(n, data)
    }
}
